packages/sql-mapper/lib/entity.js

```diff
@@ -205,13 +206,9 @@ function createMapper (db, sql, log, table, fields, primaryKey, relations, queri
       query = sql`${query} ORDER BY ${sql.join(orderBy, sql`, `)}`
     }
 
-    if (opts.limit || opts.offset !== undefined) {
-      // Use Number.MAX_SAFE_INTEGER as default value for limit because in the sql query you cannot add OFFSET without LIMIT
-      const limit = (opts.limit !== undefined) ? opts.limit : Number.MAX_SAFE_INTEGER
-      query = sql`${query} LIMIT ${limit}`
-      if (opts.offset !== undefined) {
-        query = sql`${query} OFFSET ${opts.offset}`
-      }
+    query = sql`${query} LIMIT ${sanitizeLimit(opts.limit)}`
```

Comment:
If I make a request with `limit` equals 1000 and get a successful response with 100 entities, I will think there are only 100 entities. IMO it's better to fail requests that have an exceeding `limit` param.

Plus, I would add a config option to override the max limit value.

Reply:
I see your point. 
My idea was to not impact on the user's experience by "silently" handling this scenario without throwing any kind of error, but documenting this behavior. 

In the openapi module it's also possible to leverage on the X-Total-Count header in order to not fall in the misunderstanding that you're highlighting. 
 
Anyway, as I said, I totally see your point. Let me know how to proceed for this point. 

About the config option: Ok, I'm on it!

Reply:
If limit is higher than the config we must throw an error with 400 statusCode attached.

Reply:
Implemented logic in order to : 
 - override limit's default and max value through config;
 - throw error if client ask for a limit > max accepted value. 

Anyway in this second case the error thrown by the server ([here](https://github.com/platformatic/platformatic/pull/220/files#diff-b5101b861251166c160f14838bbeba2e9c612a1bf3aecc54fbc87e73ee78cf4a)) is a 500. 
I wasn't able to find a way to return a 400 without having access to the reply object. 

How should I proceed? Currently which is the best approach in order to throw a 400 from the /sql-mapper module?